pipelines/controller/src/main/java/com/google/fhir/analytics/FlinkConfiguration.java

```diff
@@ -0,0 +1,181 @@
+      Path confPath = Files.createTempDirectory(TEMP_FLINK_CONF_DIR);
+      logger.info("Creating Flink temporary configuration directory at {}", confPath);
+      Path filePath = Paths.get(confPath.toString(), GlobalConfiguration.FLINK_CONF_FILENAME);
+      validateAndCreateConfFile(filePath, dataProperties);
+      this.flinkConfDir = confPath.toString();
+      deleteFilesRecursivelyOnExit(confPath);
+    } else {
+      String envFlinkConfDir = System.getenv(ConfigConstants.ENV_FLINK_CONF_DIR);
+      Configuration configuration;
+      if (Strings.isNullOrEmpty(envFlinkConfDir)) {
+        logger.warn(
+            "The environment variable {} is not set, hence default flink configuration will"
+                + " be used.",
+            ConfigConstants.ENV_FLINK_CONF_DIR);
+        configuration = GlobalConfiguration.loadConfiguration();
+      } else {
+        logger.info(
+            "Flink configuration will be initialised from the directory at {}", envFlinkConfDir);
+        configuration = GlobalConfiguration.loadConfiguration(envFlinkConfDir);
+      }
+      configuration = TaskExecutorResourceUtils.adjustForLocalExecution(configuration);
+      validateDirectMemoryMax(
+          configuration.get(TaskManagerOptions.NETWORK_MEMORY_MAX).getBytes(),
+          configuration.get(TaskManagerOptions.MANAGED_MEMORY_SIZE).getBytes());
+    }
+  }
+
+  private void validateAndCreateConfFile(Path filePath, DataProperties dataProperties)
+      throws IOException {
+    long networkMemoryMax = calculateNetworkMemoryMax(dataProperties);
+    validateDirectMemoryMax(networkMemoryMax, MemorySize.parseBytes(DEFAULT_MANAGED_MEMORY_SIZE));
+    try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath.toString()))) {
+      logger.info("Network memory : {}", networkMemoryMax);
+      writer.write(
+          String.format(
+              KEY_VALUE_FORMAT, TaskManagerOptions.NETWORK_MEMORY_MAX.key(), networkMemoryMax));
+      writer.newLine();
+      writer.write(
+          String.format(
+              KEY_VALUE_FORMAT,
+              TaskManagerOptions.MANAGED_MEMORY_SIZE.key(),
+              DEFAULT_MANAGED_MEMORY_SIZE));
+      writer.newLine();
+      writer.write(
+          String.format(KEY_VALUE_FORMAT, DeploymentOptions.ATTACHED.key(), Boolean.FALSE));
+      writer.newLine();
+      writer.write(
+          String.format(
+              KEY_VALUE_FORMAT,
+              DeploymentOptions.JOB_LISTENERS.key(),
+              FlinkJobListener.class.getName()));
+    }
+  }
+
+  private void validateDirectMemoryMax(long networkMemoryInBytes, long managedMemoryInBytes) {
+    long totalDirectMemory = networkMemoryInBytes + managedMemoryInBytes;
+    long totalDirectMemoryOfAllParallelPipelines =
+        totalDirectMemory * EtlUtils.NO_OF_PARALLEL_PIPELINES;
+
+    long jvmMaxDirectMemory = jdk.internal.misc.VM.maxDirectMemory();
+    if (jvmMaxDirectMemory <= totalDirectMemoryOfAllParallelPipelines) {
```

Comment:
So as discussed off-PR, let's make sure that the memory validation does not break the non-local runs, i.e., when the pipeline is running on a cluster.